dask_mongo/core.py

```diff
@@ -14,19 +15,39 @@
 
 appname = f"dask-mongo/{__version__}"
 
+_CACHE_SIZE = 10
+
+
+class CachedKwargs(dict):
+    def __hash__(self):
+        return hash(frozenset(self))
+
+
+@lru_cache(_CACHE_SIZE, typed=True)
+def _cache_inner(kwargs):
+    return pymongo.MongoClient(appname=appname, **kwargs)
+
+
+def _get_client(kwargs):
+    return _cache_inner(CachedKwargs(kwargs))
+
+
+def _get_num_clients():
+    return _cache_inner.cache_info().currsize
```

Comment:
It looks like this is only used for testing purposes. Mind moving it over to `dask_mongo/tests/test_core.py` so the definition is co-located with where it's used?